usaspending_api/etl/management/commands/load_table_from_delta.py

```diff
@@ -247,3 +293,219 @@ def handle(self, *args, **options):
+        df_no_arrays = convert_array_cols_to_string(df, is_postgres_array_format=True, is_for_csv_export=True)
+        df_no_arrays.write.options(
+            maxRecordsPerFile=CONFIG.SPARK_PARTITION_ROWS * 3,  # e.g. 25k - 50k seems a good sweet spot
+            compression="gzip",
+            nullValue=None,
+            escape='"',  # " is used to escape the 'quote' character setting (which defaults to "). Escaped quote = ""
+            ignoreLeadingWhiteSpace=False,  # must set for CSV write, as it defaults to true
+            ignoreTrailingWhiteSpace=False,  # must set for CSV write, as it defaults to true
+            timestampFormat=CONFIG.SPARK_CSV_TIMEZONE_FORMAT,
+        ).mode(saveMode="overwrite" if not keep_csv_files else "errorifexists").csv(s3_bucket_with_csv_path)
+
+        logger.debug(
+            f"Connecting to S3 at endpoint_url={CONFIG.AWS_S3_ENDPOINT}, region_name={CONFIG.AWS_REGION} to "
+            f"get listing of contents of Bucket={spark_s3_bucket_name} with Prefix={csv_path}"
+        )
+
+        if not CONFIG.USE_AWS:
+            boto3_session = boto3.session.Session(
+                region_name=CONFIG.AWS_REGION,
+                aws_access_key_id=CONFIG.AWS_ACCESS_KEY.get_secret_value(),
+                aws_secret_access_key=CONFIG.AWS_SECRET_KEY.get_secret_value(),
+            )
+            s3_resource = boto3_session.resource(
+                service_name="s3", region_name=CONFIG.AWS_REGION, endpoint_url=f"http://{CONFIG.AWS_S3_ENDPOINT}"
+            )
+        else:
+            s3_resource = boto3.resource(
+                service_name="s3", region_name=CONFIG.AWS_REGION, endpoint_url=f"https://{CONFIG.AWS_S3_ENDPOINT}"
+            )
+        s3_bucket_name = spark_s3_bucket_name
+        s3_bucket = s3_resource.Bucket(s3_bucket_name)
+        gzipped_csv_files = [f.key for f in s3_bucket.objects.filter(Prefix=csv_path) if f.key.endswith(".csv.gz")]
+        file_count = len(gzipped_csv_files)
+        logger.info(f"LOAD: Finished dumping {file_count} CSV files in {s3_bucket_with_csv_path}")
+
+        logger.info(f"LOAD: Starting SQL bulk COPY of {file_count} CSV files to Postgres {temp_table} table")
+
+        db_dsn = get_database_dsn_string()
+        partitions = 8
+        with psycopg2.connect(dsn=db_dsn) as connection:
+            with connection.cursor() as cursor:
+                cursor.execute("SHOW max_parallel_workers")
+                max_parallel_workers = int(cursor.fetchone()[0])
+                # For whatever reason, one-half of the max parallel workers seemed to be the sweet spot, but probably
+                # don't want to go below 8
+                partitions = max(ceil(max_parallel_workers / 2), partitions)
+
+        # Give more memory to each connection during the COPY operation of large files to avoid spillage to disk
+        work_mem_for_large_csv_copy = 256 * 1024  # MiB of work_mem * KiBs in 1 MiB
```

Comment:
Same suggestion as above, moving these to constants for ease of updating based on environments, etc.